### ISSUE TITLE
Make repo compatible with the webui

### ISSUE THREAD

**Issue by ljleb** [ljleb/sd-mecha#11]
We can add a python script under `scripts/mecha_extension.py` which would serve as the entry point for the webui extension. I outline here what I think we want to do, but I'm open to any suggestion really.

Must have:
- select a `.mecha` recipe in the checkpoints dropdown to load a recipe instead of a model. The recipe is merged on the spot. This allows to save a lot of disk space why also making it very easy to explore alternative model weights composition

Should have:
- create recipes from within the webui and run them. I'm not sure what interface to use for this, maybe we can just have a textbox in which we hand-craft yaml recipes

**Comment by Enferlain** [comment]
Yeah I think the biggest boost for testing would be being able to go through various recipes in batches and even iterate through them with something like an XYZ and see the results of those experiments immediately. That would already be halfway to unlocking not just executing already existing ideas but go into exploration territory. 

I feel like that is one of the biggest strengths you've carved out with Mecha, as everything else is computationally shit in comparison. Imagine being able to do a shitload of merges, more than you could manually put together, and then being able to visually confirm the results without having to manually load and gen and unload them throughout the process, and without your gpu or ram breaking a sweat.

And maybe in the future Mecha could even do the exploration part with something like [this](https://github.com/SimonBlanke/Hyperactive)

Who knows. I have suggestions but no knowledge to explain the technicals and I can't help implement them, which makes suggesting things feel a bit awkward. But I like everything so far and I'm thankful for what you e made.
